src/bin/pg_autoctl/pgctl.c

```diff
@@ -1989,7 +1989,7 @@ prepare_conninfo_sslmode(PQExpBuffer buffer, SSLOptions sslOptions)
  * the IDENTIFY_SYSTEM command to check that HBA is ready.
  */
 bool
-pgctl_identify_system(ReplicationSource *replicationSource)
+pgctl_identify_system(ReplicationSource *replicationSource, bool *unreachable)
```

Comment:
As noted in another comment I don't think that's the best way to implement that check. Let's move the ping calls to the caller.